pkgs/http_client_conformance_tests/bin/generate_server_wrappers.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Generates the '*_server_vm.dart' and '*_server_web.dart' support files.
+
+import 'dart:core';
+import 'dart:io';
+
+const vm = '''// Generated by generate_server_wrappers.dart. Do not edit.
+
+import 'package:stream_channel/stream_channel.dart';
+
+import '<server_file_placeholder>';
+
+/// Starts the redirect test HTTP server in the same process.
+Future<StreamChannel<Object?>> startServer() async {
+  final controller = StreamChannelController<Object?>(sync: true);
+  hybridMain(controller.foreign);
+  return controller.local;
+}
+''';
+
+const web = '''// Generated by generate_server_wrappers.dart. Do not edit.
+
+import 'package:stream_channel/stream_channel.dart';
+import 'package:test/test.dart';
+
+/// Starts the redirect test HTTP server out-of-process.
+Future<StreamChannel<Object?>> startServer() async => spawnHybridUri(Uri(
+    scheme: 'package',
+    path: 'http_client_conformance_tests/src/<server_file_placeholder>'));
```

Comment:
Can something be done to make sure this generated line is not too long, won't require reformatting before checking-in?
Or even better solution would be not to check them in at all and use https://pub.dev/packages/source_gen instead to generate them as part of the build.

Reply:
Not all packages that consume this will be using the build system already. I don't think we should impose it for now.

> won't require reformatting before checking-in?

We can add a `dart format` step in this script if this situation comes up. I don't think we need to be too particular about this - it isn't a process we're publishing for others to adopt, just a quick hack to make this pattern painless locally.

Reply:
I'm happy to use Dart format now... it seems trivial to do. I'll report back when done.

Reply:
Done.